ansible/library/port_alias.py

```diff
@@ -101,33 +123,64 @@ def get_portmap(self):
                             alias_index = index
                         if 'speed' in text:
                             speed_index = index
+                        if 'role' in text:
+                            role_index = index
             else:
                 if re.match('^Ethernet', line):
                     mapping = line.split()
                     name = mapping[0]
+                    if (role_index != -1) and (len(mapping) > role_index):
+                        role = mapping[role_index]
+                    else:
+                        role = 'Ext'
                     if alias_index != -1 and len(mapping) > alias_index:
                         alias = mapping[alias_index]
                     else:
                         alias = name
-                    aliases.append(alias)
-                    portmap[name] = alias
-                    aliasmap[alias] = name
-                    if (speed_index != -1) and (len(mapping) > speed_index):
-                        portspeed[alias] = mapping[speed_index]
+                    if (all_ports == 'False' and role == 'Ext') or \
+                       (all_ports == 'True'):
```

Comment:
When `all_ports='False'`, it means that this module will not parse information of all the ports in port_config.ini, but just the external ports, right? What if the port_config.ini file is like below and `all_ports='False'`:
```
# name          lanes          alias            asic_port_name   role
Ethernet0     33,34,35,36      Ethernet1/1      Eth0-ASIC0        E
Ethernet4     29,30,31,32      Ethernet1/2      Eth1-ASIC0        E
Ethernet8     41,42,43,44      Ethernet1/3      Eth2-ASIC0        E
Ethernet12    37,38,39,40      Ethernet1/4      Eth3-ASIC0        E
Ethernet-BP0  13,14,15,16      Ethernet-BP0     Eth4-ASIC0        I
Ethernet-BP4  17,18,19,20      Ethernet-BP4     Eth5-ASIC0        I
Ethernet-BP8  21,22,23,24      Ethernet-BP8     Eth6-ASIC0        I
Ethernet-BP12 25,26,27,28      Ethernet-BP12    Eth7-ASIC0        I 
```
The above example is taken from https://github.com/Azure/SONiC/pull/644
The `role` variable may have value `E` or `I`. According to the condition here, all the interfaces will be skipped. Is this intended?

Reply:
@wangxin Thank you for pointing out the example in the document, will update it there. The role will have Ext/Int. The changes for virtual switch was added using this PR: https://github.com/Azure/sonic-buildimage/pull/5121
Yes, you are right. port_alias is called during the initial testbed set up, to get the list of interfaces with which the virtual switch has to come up. During this invocation of port_alias, we need to get only external interfaces, which is done by "all_ports=False". If port_config.ini does not have "role" column, then all ports will be considered as external interface which will be the case for single asic platform.

Reply:
Then it makes sense to me. The example in the document needs to be updated.

Reply:
Update document Azure/SONiC#644 to reflect Int/Ext role values in port_config.ini.

Reply:
@SuvarnaMeenakshi Could you help resolve the conflicts?